### PR TITLE
Electron move files from download to current directory

### PR DESCRIPTION
**Pull request by AhsanAyaz**
# Description

Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.

Fixes # (issue)

## Type of change

Please delete options that are not relevant.

- [x] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected)

# Checklist:

- [ ] I have performed a self-review of my own code
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have run tests (npm run test & npm run e2e) that prove my fix is effective or that my feature works

### CODE REVIEW

File: .vscode/extensions.json [hunk 1]
```diff
@@ -0,0 +1,3 @@
+{
```
Comment (AhsanAyaz) [inline .vscode/extensions.json:1]: This file should not be committed. Isko .gitignore file may daldo. In fact, pura .vscode ka folder gitignore hoga

File: app/main.ts [hunk 2]
```diff
@@ -1,11 +1,14 @@
 import { app, BrowserWindow, ipcMain, screen } from "electron";
 import * as path from "path";
 import * as url from "url";
+import * as fs from "fs-extra";
 
 // Initialize remote module
 require("@electron/remote/main").initialize();
 
 let win: BrowserWindow = null;
+const userDirectory =
```
Comment (AhsanAyaz) [inline app/main.ts]: Move this whole variable into a separate file, call it electron-config.ts, import the variable here then

File: app/main.ts [hunk 3]
```diff
@@ -20,6 +23,7 @@ function createWindow(): BrowserWindow {
     width: size.width,
     height: size.height,
     webPreferences: {
+      webSecurity: false,
```
Comment (AhsanAyaz) [inline app/main.ts:25]: Why do we need this?

File: app/main.ts [hunk 4]
```diff
@@ -60,7 +64,14 @@ try {
   // initialization and is ready to create browser windows.
   // Some APIs can only be used after this event occurs.
   // Added 400 ms to fix the black background issue while using transparent window. More detais at https://github.com/electron/electron/issues/15947
-  app.on("ready", () => setTimeout(createWindow, 400));
+  app.on("ready", () => {
+    setTimeout(createWindow, 400);
+    if (!fs.existsSync(userDirectory)) {
```
Comment (AhsanAyaz) [inline app/main.ts:68]: 👍🏼

File: app/main.ts [hunk 5]
```diff
@@ -79,9 +90,20 @@ try {
     }
   });
 
-  ipcMain.on("asynchronous-message", (event, arg) => {
-    console.log("copying files");
-    event.reply("asynchronous-reply", "pong");
+  ipcMain.on("check-files-are-created", (event, arg) => {
```
Comment (AhsanAyaz) [inline app/main.ts]: Fix the linting errors mentioned here.

File: app/main.ts [hunk 6]
```diff
@@ -79,9 +90,20 @@ try {
...
+    try {
+      setInterval(() => {
+        if (
+          fs.existsSync(userDirectory + "/model.json") &&
```
Comment (AhsanAyaz) [inline app/main.ts:99]: There's a case where we already have those files created. In which case, we should always make sure that the directory is empty. Or delete the files. Then we should be saving them.

File: src/app/collecting/collecting.component.ts [hunk 7]
```diff
@@ -0,0 +1,191 @@
...
+      epochs: 50,
+    };
+    this.brain.train(options, this.finishedTraining.bind(this));
+  }
+
+  // This function creates the models files and then navigate to the home page
+  finishedTraining() {
+    alert("please save the files on Downloads/ng-posture-buddy/model");
+    this.brain.save();
+    ipcRenderer.once("file-created", (event, arg) => {
```
Comment (AhsanAyaz) [inline src/app/collecting/collecting.component.ts]:
```suggestion
    ipcRenderer.once("files-created", (event, arg) => {
```

Comment (AhsanAyaz) [reply src/app/collecting/collecting.component.ts]: Change it on the other side too

File: src/app/collecting/collecting.component.ts [hunk 8]
```diff
@@ -0,0 +1,191 @@
...
+        }
+        const posture = [this.position];
+        this.brain.addData(inputs, posture);
+      }
+    }
+  }
+
+  // This function shows the poseNet is ready
+  modelLoaded() {
+    console.log("poseNet ready");
```
Comment (AhsanAyaz) [inline src/app/collecting/collecting.component.ts]: Remove this function or log. We don't need it anymore I think

File: src/app/home/home.component.ts [hunk 9]
```diff
@@ -1,22 +1,167 @@
...
+  // This function get the cordinates of body and set it into the pose varaible
+  gotPoses(poses: string | any[]) {
+    if (poses.length > 0) {
+      this.pose = poses[0].pose;
+    }
+  }
+
+  // This function is called when webcam is show on page
+  modelLoaded() {
+    console.log("poseNet ready");
```
Comment (AhsanAyaz) [inline src/app/home/home.component.ts]: We don't need this I believe

File: .vscode/extensions.json (hunk 1 above)
Comment (mohsinayaz) [reply .vscode/extensions.json:1]: done

File: app/main.ts (hunk 2 above)
Comment (mohsinayaz) [reply app/main.ts]: done

File: app/main.ts (hunk 3 above)
Comment (mohsinayaz) [reply app/main.ts:25]: because application not allowed to get model files from download directory

File: app/main.ts (hunk 4 above)
Comment (mohsinayaz) [reply app/main.ts:68]: done

File: app/main.ts (hunk 6 above)
Comment (mohsinayaz) [reply app/main.ts:99]: done

**Comment by AhsanAyaz** [comment]
Thanks @mohsinayaz 👍🏼 Good work!

**Pull request merged**